braindecode/datasets/base.py

```diff
@@ -0,0 +1,62 @@
+"""
+Dataset classes.
+"""
+
+# Authors: Hubert Banville <hubert.jbanville@gmail.com>
+#          Lukas Gemein <l.gemein@gmail.com>
+#          Simon Brandt <simonbrandt@protonmail.com>
+#          David Sabbagh <dav.sabbagh@gmail.com>
+#
+# License: BSD (3-clause)
+
+from torch.utils.data import Dataset
+
+
+class BaseDataset(Dataset):
+    """
+    A base dataset.
+
+    Parameters
+    ----------
+    raw: mne.Raw
+    info: pandas.DataFrame
+        holds additional information about the raw
```

Comment:
What is the expected shape of the DataFrame? I don't see where this is being used right now and I'm not sure what it is supposed to be containing.

Reply:
Shape of DataFrame is 1 (row) x n_infos (cols). I took a photo of the flipchart in Inria where you guys designed the dataset classes. There was a BaseDataset which contains information on a dataset level. (There was also a EpochsDataset that contains information on a sample level. Or as I know interpreted it, on a windows/supercrop level, since windows/supercrops can have the size of 1 sample in the extrema.) For example, for MOABB datasts the BaseDataset info can be subject_id, session_id, and run_id. For the Temple University dataset, this can be the age and gender of patients as well as the pathology status of EEG recordings. (The WindowsDataset info will then be i_supercrop_intrial, i_start_of_supercrop, i_stop_of_supercrop, which are required for correctly combining supercrop/window predictions as well as the supercrop/window target.)

Reply:
Ah I get it, this makes sense! What do you think of using a simple dict instead of a DataFrame here?

Reply:
The WindowsDataset adopts (or concats) the info DataFrame from its base datasets (see https://github.com/robintibor/braindecode-1/blob/dataset-integration/braindecode/datasets/datasets.py#L236, https://github.com/robintibor/braindecode-1/blob/dataset-integration/braindecode/datasets/datasets.py#L360)
It can be used to conveniently create groups within the WindowsDataset based on pandas groupby function (which you also wrote down on the flipchart :)). See <code> plot_dataset_example.py</code> for an example!

Reply:
Yes, I remember that now. :P 

1- It is a small implementation detail then, but we might as well discuss it: since the info dataframe only contains one row in BaseDataset, it could be a simple dict (simplifies the code a bit). Is there a case where a DataFrame might be preferable, knowing that concatenating from dicts is as simple as if not simpler then it is with DataFrames?

2- As for the WindowsDataset, where do you see the groupby functionality happening? Would that be on the internal metadata attribute of the Epochs object? In that case, if I understand correctly, we would need to add columns to the metadata object with 'age', 'sex', 'pathological', etc.

Reply:
@gemeinl Reading another one of your answers below made me think: is it because we potentially want to concatenate BaseDatasets directly without passing through mne.Epochs? In that case I would understand the point for a DataFrame (although this might be covered already by the length-1 epoch case we tested during the sprint).